roles/splunk/tasks/kvstore_upgrade.yml

```diff
@@ -0,0 +1,156 @@
+---
+- name: Check if authenticated
+  include_tasks: splunk_login.yml
+  when: not splunk_authenticated
+
+- name: Set fact for kvstore Upgrade
+  ansible.builtin.set_fact:
+    splunk_upgrade_server: false
+
+- name: Check the current kvstore storage backend
+  ansible.builtin.command: |
+    {{ splunk_home }}/bin/splunk show kvstore-status | grep storageEngine | sed -r 's/\s+storageEngine : //g'
+  become: true
+  become_user: "{{ splunk_nix_user }}"
+  register: splunk_kvstore_backend_out
+  changed_when: splunk_kvstore_backend_out.rc == 0
+  failed_when: splunk_kvstore_backend_out.rc != 0
+
+- name: Check the current kvstore server version
+  ansible.builtin.command: |
+    {{ splunk_home }}/bin/splunk show kvstore-status | grep serverVersion | sed -r 's/\s+serverVersion : //g'
+  become: true
+  become_user: "{{ splunk_nix_user }}"
+  register: splunk_current_server_version_out
+  changed_when: splunk_current_server_version_out.rc == 0
+  failed_when: splunk_current_server_version_out.rc != 0
+
+- name: Debug print kvstore backend engine
+  ansible.builtin.debug:
+    var: splunk_kvstore_backend_out.stdout
+    verbosity: 2
+
+- name: Debug print kvstore backend engine
+  ansible.builtin.debug:
+    var: splunk_current_server_version_out.stdout
+    verbosity: 2
+
+- name: Upgrade KVstore if needed
+  block:
+    - name: Backup KVStore
+      include_tasks: adhoc_backup_kvstore.yml
+      vars:
+        - archive_name: "-archiveName preAnsibleVersionUpgradeBackup"
```

Comment:
Should this maybe be customizable?

Reply:
Eh, not in my opinion, but it's trivial to do so.